streaming/src/queue/streaming_queue.h

```diff
@@ -0,0 +1,217 @@
+#ifndef _STREAMING_QUEUE_H_
+#define _STREAMING_QUEUE_H_
+#include <iterator>
+#include <list>
+#include <vector>
+
+#include "ray/common/id.h"
+#include "ray/util/util.h"
+
+#include "streaming_logging.h"
+#include "streaming_queue_item.h"
+#include "transport.h"
+#include "utils.h"
+
+namespace ray {
+namespace streaming {
+
+using ray::ObjectID;
+
+enum QueueType { UPSTREAM = 0, DOWNSTREAM };
+
+/// A queue-like data structure, which do not delete it's item
+/// after poped. The lifecycle of each item is:
+/// - Pending, a item is pushed into queue, but has not been
+///   processed (sent out or consumed),
+/// - Processed, has been handled by user, but should not be deleted
+/// - Evicted, useless to user, should be poped and destroyed.
+/// At present, this data structure is implemented with one std::list,
+/// using a watershed iterator to divided.
+class Queue {
+ public:
+  /// \param size max size of the queue in bytes.
+  Queue(uint64_t size, std::shared_ptr<Transport> transport)
+      : max_data_size_(size), data_size_(0), data_size_sent_(0) {
+    buffer_queue_.push_back(NullQueueItem());
+    watershed_iter_ = buffer_queue_.begin();
+  }
+
+  virtual ~Queue() {}
+
+  /// Push item into queue, return false is queue is full.
```

Comment:
```suggestion
  /// Push an item into the queue. 
  /// \return false if the queue is full.
```

We use this style for C++ function documentation.

Reply:
> /// \return false if the queue is full.

fixed.